service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/C29RecoveryOfAChildDocumentParameterGeneratorTest.java

```diff
@@ -0,0 +1,303 @@
+
+        DocmosisParameters generatedParameters = underTest.generate(caseData);
+        DocmosisParameters expectedParameters = expectedCommonParameters()
+            .orderDetails(getStandardMessage(selectedChildren.size(), PlacedUnderOrder.CARE_ORDER)
+                + getRemoveMessage()
+                + getIsExparteMessage(false))
+            .orderHeader(ORDER_HEADER)
+            .orderMessage(ORDER_MESSAGE)
+            .build();
+
+        assertThat(generatedParameters).isEqualTo(expectedParameters);
+    }
+
+    @Test
+    void shouldGenerateDocumentWithAllActionsPermittedOfficerNameEnteredAndIsExParte() {
+        CaseData caseData = CaseData.builder()
+            .caseLocalAuthority(LA_CODE)
+            .caseLocalAuthorityName(LA_NAME)
+            .manageOrdersEventData(ManageOrdersEventData.builder()
+                .manageOrdersOrderCreatedDate(ORDER_CREATED_DATE)
+                .manageOrdersFurtherDirections(FURTHER_DIRECTIONS)
+                .manageOrdersActionsPermittedAddress(REMOVAL_ADDRESS)
+                .manageOrdersOfficerName(OFFICER_NAME)
+                .manageOrdersActionsPermitted(List.of(
+                    C29ActionsPermitted.ENTRY,
+                    C29ActionsPermitted.INFORM,
+                    C29ActionsPermitted.PRODUCE,
+                    C29ActionsPermitted.REMOVE
+                ))
+                .manageOrdersPlacedUnderOrder(PlacedUnderOrder.EMERGENCY_PROTECTION_ORDER)
+                .manageOrdersIsExParte("Yes")
+                .build())
+            .build();
+
+        List<Element<Child>> selectedChildren = wrapElements(child);
+
+        when(childrenService.getSelectedChildren(caseData)).thenReturn(selectedChildren);
+
+        DocmosisParameters generatedParameters = underTest.generate(caseData);
+        DocmosisParameters expectedParameters = expectedCommonParameters()
+            .orderDetails(getStandardMessage(selectedChildren.size(),
+                PlacedUnderOrder.EMERGENCY_PROTECTION_ORDER)
+                + getEntryMessage(selectedChildren.size())
+                + getInformMessage(caseData.getManageOrdersEventData())
+                + getProduceMessage()
+                + getRemoveMessage()
+                + getIsExparteMessage(true))
+            .orderHeader(ORDER_HEADER)
+            .orderMessage(ORDER_MESSAGE)
+            .build();
+
+        assertThat(generatedParameters).isEqualTo(expectedParameters);
+    }
+
+    private C29RecoveryOfAChildDocmosisParametersBuilder<?,?> expectedCommonParameters() {
+        return C29RecoveryOfAChildDocmosisParameters.builder()
+            .furtherDirections(FURTHER_DIRECTIONS)
+            .localAuthorityName(LA_NAME)
+            .dateOfIssue(FORMAT_LOCAL_DATE_TO_STRING)
+            .orderTitle(C29_RECOVERY_OF_A_CHILD.getTitle());
+    }
+
+    private String getStandardMessage(int numberOfChildren, PlacedUnderOrder order) {
```

Comment:
I'd say the same for all of the similar methods.

Reply:
will update